packages/hash/frontend/src/pages/[account-slug]/entities/new.page.tsx

```diff
@@ -0,0 +1,68 @@
+import { useRouter } from "next/router";
+import { useState } from "react";
+import { useEffectOnceWhen } from "rooks";
+import { useLoggedInUser } from "../../../components/hooks/useAuthenticatedUser";
+import { PageErrorState } from "../../../components/page-error-state";
+import { getPlainLayout, NextPageWithLayout } from "../../../shared/layout";
+import { mustBeVersionedUri } from "../types/entity-type/util";
+import { EntityPageLoadingState } from "./[entity-uuid].page/entity-page-loading-state";
+import { NewEntityPage } from "./[entity-uuid].page/new-entity-page";
+import { useCreateNewEntityAndRedirect } from "./[entity-uuid].page/shared/use-create-new-entity-and-redirect";
+
+type PageState = "error" | "loading" | "ready";
+
+const Page: NextPageWithLayout = () => {
+  const router = useRouter();
+  const { authenticatedUser } = useLoggedInUser();
+  const createNewEntityAndRedirect = useCreateNewEntityAndRedirect();
+  const [pageState, setPageState] = useState<PageState>("loading");
+
+  useEffectOnceWhen(() => {
```

Comment:
I think things like `useEffectOnceWhen` fundamentally misunderstand React effects. They're not *for* event callbacks, and it makes things muddy when you try to use them as such. 

A better way of handlnig this is to update `useLoggedInUser` to allow you to call a callback on the event of the user loading.

Reply:
> A better way of handling this is to update `useLoggedInUser` to allow you to call a callback on the event of the user loading.

I totally agree with this. But I don't think `useEffectOnceWhen` is that much horrible, as long as used with a clear condition. Here, it just represents "run this init function when user is fetched". We can create a ticket and refactor `useLoggedInUser` to support callback I guess, but I feel like this should be ok since it's really simle, and works as expected.

Reply:
I actually feel fairly strongly it should happen in this PR. This is an abuse of effects – it's requiring on an intermediary re-render. It wouldn't be too difficult to add a callback now.

Reply:
Implemented here https://github.com/hashintel/hash/pull/1492/commits/96b5bbda8020c4e9ccea1178af93cf4f95f5727e